spec/support/webmock_ruby24_bridge.rb

```diff
@@ -0,0 +1,15 @@
+# Ruby 2.4.0 removed the closed? check in the conditional in: s.close if !s.closed?
+# Webmock was changed to add close to StubSocket along with another change.
+# https://github.com/ruby/ruby/commit/f845a9ef76c0195254ded79c85c24332534f4057
+# https://github.com/bblimke/webmock/commit/8f2176a1fa75374df55b87d782e08ded673a75b4
+# WebMock 2.3.1+ fixed this.
+# We should upgrade webmock but that requires some re-recording of cassettes (I think)
+# and maybe other things.
+if WebMock::VERSION < "2.3.1"
+  class StubSocket
+    def close
+    end
+  end
+else
+  warn "Remove me: #{__FILE__}:#{__LINE__}. WebMock 2.3.1+ fixed the issue with ruby 2.4.0 by adding StubSocket#close."
```

Comment:
Yeah, also, re-recording cassettes is a major bloat on a repo if everything is replaced completely.  Might be better if we can figure out how to just update the yaml of the cassettes in a programmatic fashion with the needed changes for webmock to avoid needing to actually re-record the cassettes.